src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -106,6 +106,53 @@ internal IEnumerator<TElement> GetEnumerator(int minIdx, int maxIdx)
         }
     }
 
+    internal sealed class OrderedKeylessEnumerable<TElement> : OrderedEnumerable<TElement>
```

Comment:
Honestly, I'm not convinced all this added code is justified simply to avoid using an `x => x` key selector. It would be interesting to see how this performs in benchmarks compared to `OrderBy(x => x)`. Given the comparison chaining apparatus that `Order` still needs to support I'd be surprised if eliminating the key selector even registers perf-wise.

cc @stephentoub

Reply:
The problem is that delegates and lambdas are currently not optimized at all by the JIT. Therefore, I thought it would be best to avoid them. In my implementation I also use one at one point. 
However, we should also note that LINQ should never be considered a high-performance API and thus it probably wouldn't be that bad.
See <https://github.com/dotnet/runtime/issues/63425#issuecomment-1006230403>

Reply:
That's true, but I wouldn't think it would constitute a dominant factor in this particular implementation.

Reply:
I can do some benchmarks.

Another thing worth mentioning regarding the delegates is that #68703 probably limits or even completely improves this problem.

Reply:
I couldn't tell much difference now, since I'm effectively creating the delegate just one level down.
See benchmarks: [rt-issue-67194](https://github.com/deeprobin/performance/tree/rt-issue-67194)

Therefore I have now changed the whole thing to a redirect to OrderBy.